### PR TITLE
Update to .NET 9 rc1

### PR DESCRIPTION
**Pull request by vzarytovskii**

### CODE REVIEW

**Comment by vzarytovskii** [comment]
I have fixed most of issues, but there's currently no way around of using shipped FSharp.Build (and inherently FSharp.Core project).

**Comment by Martin521** [comment]
Locally, everything still works fine for me.
[I have seen these nullness errors before](https://github.com/dotnet/fsharp/pull/17542#issuecomment-2291263607), but never was able to repro them reliably.

**Comment by vzarytovskii** [comment]
> Locally, everything still works fine for me. [I have seen these nullness errors before](https://github.com/dotnet/fsharp/pull/17542#issuecomment-2291263607), but never was able to repro them reliably.

It is not yet merged. This PR was breaking it by default, i "fixed" it, but it will use fsharp.core project by default as opposed to library.

**Comment by vzarytovskii** [comment]
Either way, I think rc1 update will have to wait, we're hitting
https://github.com/dotnet/core/pull/9489

File: Directory.Build.props
Comment (vzarytovskii) [inline Directory.Build.props:52]: That'd be a "workaround" to unblock plain builds, but it will by default use the project.

File: FSharp.Profiles.props
Comment (vzarytovskii) [inline FSharp.Profiles.props:28]: Turning it off currently for the plain builds, so we don't get mismatches with defines we have.

File: tests/FSharp.Compiler.ComponentTests/Interop/ByrefTests.fs
Comment (vzarytovskii) [inline tests/FSharp.Compiler.ComponentTests/Interop/ByrefTests.fs:137]: Not an extension anymore

**Review by Thompson1985 — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

File: eng/Build.ps1
Comment (vzarytovskii) [inline eng/Build.ps1:74]: After talking to @T-Gro, for shipping release, we decided to ship compiler with turned off realsig to avoid any potential optimization issues is compiler and FCS themselves.

It's ultimately up to @KevinRansom to approve this part or reject (so I will revert this change).

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**
